lib/facility_factory.rb

```diff
@@ -0,0 +1,18 @@
+class FacilityFactory
+  def initialize
+    
+  end
+
+  def create_facility(dmv)
```

Comment:
I think that this method is a great starting point. And yes, using these `||` conditions can feel a bit clunky especially if we were to add more datasets. So one thing that you can consider is creating some helper methods in here that take care of grabbing the data that you want based on how it is formatted. So an example could be:
```
def format_address(data)
   if data[:website] && data[:website].include?('oregon')
      address = JSON.parse(data[:location_1][:human_address], symbolize_names: true)
      "#{address[:address]}, #{address[:city]}, #{address[:state]} #{address[:zip]}"
    elsif data[:state] && data[:state] == 'NY'
      "#{data[:street_address_line_1]}, #{data[:city]}, #{data[:state]} #{data[:zip_code]}"
    else data[:state] && data[:state] == 'missouri'
      "#{data[:address1]}, #{data[:city]}, #{data[:state]}, #{data[:zipcode]}"
    end
end
```
This would help with formatting the addresses based on how they come in via the external data source. You could create additional helpers for name, phone_number. You could then create another helper method that calls on all of the other helpers:
```
def format_facility_data(facility)
    data = {
      name: format_name(facility),
      address: format_address(facility),
      phone: format_phone_number(facility)
    }
end
```
You would then call this method inside of your `create_facilities` method. Let me know if you have questions.